src/hooks/charts/useChartInfo.ts

```diff
@@ -7,75 +7,73 @@ import { createSelector } from 'reselect';
-    charts,
-    fetchingCharts,
-  }),
-  (_: any, address: any) => address,
-  (state, { address, chartType }) => {
-    const { charts, fetchingCharts } = state;
-    const chartsForAsset = {
-      ...charts?.[address],
-    };
-    return {
-      chart: formatChartData(chartsForAsset?.[chartType]),
-      chartsForAsset,
-      fetchingCharts,
-    };
-  }
-);
-
-export default function useChartInfo(asset: any) {
-  const dispatch = useDispatch();
-  const { setParams } = useNavigation();
-
-  const { params } = useRoute<{
-    key: string;
-    name: string;
-    params: any;
-  }>();
-  const { address } = asset;
-
-  const chartType = params?.chartType ?? DEFAULT_CHART_TYPE;
+const chartTimes = ['hour', 'day', 'week', 'month', 'year'] as const;
+type ChartTime = typeof chartTimes[number];
+type PriceChartTimeData = { points?: [x: number, y: number][] };
 
-  const { chart, chartsForAsset, fetchingCharts } = useSelector(
-    // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '(state: never) => { chart: any; ... Remove this comment to see the full error message
-    useCallbackOne(state => chartSelector(state, { address, chartType }), [
-      address,
-      chartType,
-    ]),
-    isEqual
+const getChartTimeArg = (selected: ChartTime) =>
+  chartTimes.reduce(
+    (args, time) => ({ ...args, [time]: time === selected }),
+    {} as Record<ChartTime, boolean>
   );
 
-  useEffect(() => {
-    if (!disableCharts) {
-      dispatch(emitChartsRequest(address, chartType));
-    }
-  }, [address, chartType, dispatch]);
+export type ChartData = { x: number; y: number };
 
+const fetchPriceChart = async (
+  time: ChartTime,
+  chainId: NetworkProperties['id'],
+  address: string
+) => {
+  const priceChart = await metadataClient
+    .priceChart({ address, chainId, ...getChartTimeArg(time) })
+    .then(d => d.token?.priceCharts[time] as PriceChartTimeData);
+  return priceChart?.points?.reduce((result, point) => {
+    result.push({ x: point[0], y: point[1] });
+    return result;
+  }, [] as ChartData[]);
+};
+
+export const usePriceChart = ({
+  mainnetAddress,
+  address,
+  network,
+}: {
+  mainnetAddress?: string;
+  address: string;
+  network: Network;
+}) => {
+  const { setParams } = useNavigation();
   const updateChartType = useCallback(
     (type: any) => {
       setParams({ chartType: type });
     },
     [setParams]
   );
-
-  return {
-    chart,
-    charts: chartsForAsset,
-    chartType,
-    fetchingCharts,
-    updateChartType,
-  };
-}
+  const { params } = useRoute<{
+    key: string;
+    name: string;
+    params: any;
+  }>();
+  const chartType = params?.chartType ?? DEFAULT_CHART_TYPE;
+  const chainId = getNetworkObj(network).id;
+  const mainnetChainId = getNetworkObj(Network.mainnet).id;
+  const query = useQuery({
+    queryFn: async () => {
+      const chart = await fetchPriceChart(chartType, chainId, address);
+      if (!chart && mainnetAddress)
```

Comment:
did you see an instance of this happening? i think this should be totally handled by BE with just address & network

cc @welps any reason for us to need a back up here?

Reply:
no, i copied that from BX cc @derHowie

Reply:
We can handle this, but nobody talked to me about it

Which charts don't work?

Reply:
We can handle this, but nobody talked to me about it

Which charts don't work?